visualization/app/codeCharta/core/settings/settings.service.ts

```diff
@@ -294,12 +330,12 @@ export class SettingsService implements DataServiceSubscriber, CameraChangeSubsc
         this._settings.margin = settings.margin;
         this._settings.deltas = settings.deltas;
         this._settings.operation = settings.operation;
-        this._settings.deltaColorFlipped = this.settings.deltaColorFlipped;
+        this._settings.deltaColorFlipped = settings.deltaColorFlipped;
         this._settings.minimizeDetailPanel = settings.minimizeDetailPanel;
         this._settings.invertHeight = settings.invertHeight;
 
         //TODO what to do with map ? should it even be a part of settings ? deep copy of map ?
-        this._settings.map = this.settings.map;
```

Comment:
using this.settings.map ignored the given map in the settings object. This allowed to set the settings without changing a map (used in scenarios! They do not provide a map). Setting a scenario prevents your implementation to work. Please adjust the scenario application to your methods.

Reply:
Thanks for the info, line 302 was corrected from 

        this._settings.map = settings.map;
to 
        this._settings.map = settings.map|| this.settings.map;